addons/point_of_sale/models/stock_move.py

```diff
@@ -0,0 +1,6 @@
+from odoo import api, fields, models
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    pos_order_id = fields.Many2one(comodel_name='pos.order', string="Pos Order")
```

Comment:
I don't think you need it as there is a picking linked to the pos order. On order you can take oreder.picking_id.move_lines or on move you can take move.picking_id == order.picking_id